src/Stratis.Bitcoin.Features.Wallet/MultisigTransactionHandler.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using NBitcoin.Policy;
+using Stratis.Bitcoin.Features.Wallet.Interfaces;
+using Stratis.Bitcoin.Features.Wallet.Models;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.Wallet
+{
+    /// <summary>
+    /// A handler that has various functionalities related to transaction operations.
+    /// </summary>
+    public class MultisigTransactionHandler : WalletTransactionHandler, IMultisigTransactionHandler
+    {
+        private readonly ILogger logger;
+
+        private readonly Network network;
+        
+        private readonly StandardTransactionPolicy transactionPolicy;
+
+        private readonly IWalletManager walletManager;
+
+        private readonly IWalletFeePolicy walletFeePolicy;
+
+        public MultisigTransactionHandler(
+            ILoggerFactory loggerFactory,
+            IWalletManager walletManager,
+            IWalletFeePolicy walletFeePolicy,
+            Network network,
+            StandardTransactionPolicy transactionPolicy)
+            : base(loggerFactory, walletManager, walletFeePolicy, network, transactionPolicy)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+        }
+
+        /// <inheritdoc />
+        public Transaction BuildTransaction(TransactionBuildContext context, SecretModel[] secrets)
+        {
+            if (secrets == null || secrets.Length == 0)
+               throw new WalletException("Could not build the transaction. Details: no private keys provided");
+
+            this.InitializeTransactionBuilder(context);
+
+            if (context.Shuffle)
+                context.TransactionBuilder.Shuffle();
+
+            Transaction unsignedTransaction = context.TransactionBuilder.BuildTransaction(false);
+            var signedTransactions = new List<Transaction>();
+            foreach (SecretModel secret in secrets)
```

Comment:
I like the idea of a multisig API endpoint in general, but I am not sure this approach is actually going to be useful in typical scenarios. Normally, a node will only have one of the private keys available to it, and the partially signed transaction must be communicated to the other signers in some way (this is not via transaction relay, i.e. it must be done out-of-band).

Reply:
What you could perhaps do is have a list of pubkeys passed into the API endpoint, and generate the P2SH scriptPubKey from there. You would need another method to return the redeemScript for a given pubkey list as well I think. Or if it's bare multisig instead of P2SH multisig, the scriptPubKey and scriptSig generation looks somewhat different again.